miscellaneous_scripts/extract_apt_patch_data.py

```diff
@@ -0,0 +1,172 @@
+    Thereafter, it sends the relevant data to process_packages method to find out all the impacted packages that can be upgraded.
+    In the ends, it dumps the data at save_result_path location.
+    """
+    impacted_packages = args.impacted_packages
+    assert impacted_packages, "Impacted packages need to be passed for generative mode."
+    installed_packages = get_package_list_using_command(run_command="apt list --installed")
+    upgradable_packages = get_package_list_using_command(run_command="apt list --upgradable")
+
+    upgradable_packages_data_for_impacted_packages = {}
+    patch_package_list = []
+    patch_package_dict = {}
+
+    process_packages(
+        installed_packages,
+        impacted_packages,
+        upgradable_packages,
+        patch_package_list,
+        upgradable_packages_data_for_impacted_packages,
+    )
+
+    patch_package_dict = get_installed_version_for_packages(patch_package_list)
+    for k, _ in patch_package_dict.items():
+        patch_package_dict[k]["previous_version"]=patch_package_dict[k].pop("installed_version")
+
+    apt_patch_details = {
+        "patch_package_dict": patch_package_dict,
+        "upgradable_packages_data_for_impacted_packages": upgradable_packages_data_for_impacted_packages
+    }
+    with open(args.save_result_path, "w") as outfile:
+        json.dump(apt_patch_details, outfile, indent=4)
+
+
+def process_modify_mode_type(args):
+    with open(args.save_result_path, "r") as readfile:
+        json_data = json.load(readfile)
+    patch_package_list = list(json_data["patch_package_dict"].keys())
+    current_patch_package_dict = get_installed_version_for_packages(patch_package_list)
+    for k,_ in json_data["patch_package_dict"].items():
+        json_data["patch_package_dict"][k].update(current_patch_package_dict[k])
+    with open(args.save_result_path, "w") as outfile:
+        json.dump(json_data, outfile, indent=4)
+
+
+def main():
+    """
+    This script takes in the list of impacted packages and finds all the apt packages that need to be and can be upgraded for patching.
+    The script can work in 2 modes - genrate and modify. During the generate mode, the patchable packages are found out and are stored
+    with their current version at the save-result-path location. During the modify phase, the patchable packages are read from the
+    save-result-path location and their latest version is stored at the save-result-path location itself. The GENERATE mode is run on
+    the released DLCs to find out the patchable packages and their version in the released DLCs. Once the released DLCs are patched,
+    the same script is run in the MODIFY mode to add the latest version data of the patched packages.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--mode_type", type=str, choices=[ModeType.GENERATE, ModeType.MODIFY], required=True)
```

Comment:
There is a way to extract all the Enum values as a list. Using that would make it future scalable

Reply:
Done